### PR TITLE
[test] Re-add crash tests for getObjects:andKeys:count:

### PR DESCRIPTION
**Pull request by lorentey**
On some platforms, when a new process is started, ARC’s autoreleased return value optimization is expected to fail the first time it is used in each linked dylib. StdlibUnittest takes care of warming up ARC for the stdlib (libswiftCore.dylib), but for Dictionary.swift, we also need to do it for Foundation, or there will be spurious leaks reported for tests immediately following a crash test.

Add the necessary dummy operations to setUp, and re-add the crash tests that were removed in #17862. (As separate tests this time, so they don’t hide leaks in non-crashing test parts.)

rdar://problem/42069800

### CODE REVIEW

**Comment by lorentey** [comment]
cc @moiseev

**Comment by lorentey** [comment]
@swift-ci please test

File: validation-test/stdlib/Dictionary.swift [hunk 1]
```diff
@@ -4751,6 +4763,22 @@ DictionaryTestSuite.test("Hashable") {
...
 DictionaryTestSuite.setUp {
+#if _runtime(_ObjC)
+  // Exercise ARC's autoreleased return value optimization in Foundation.
+  //
+  // On some platforms, when a new process is started, the optimization is
+  // expected to fail the first time it is used in each linked
+  // dylib. StdlibUnittest takes care of warming up ARC for the stdlib
+  // (libswiftCore.dylib), but for this particular test we also need to do it
+  // for Foundation, or there will be spurious leaks reported for tests
+  // immediately following a crash test.
```
Comment (moiseev) [inline validation-test/stdlib/Dictionary.swift]: 😱

**Comment by lorentey** [comment]
Linux failure was in lldb; let's retry for good luck.

@swift-ci please clean smoke test linux platform

File: validation-test/stdlib/Dictionary.swift (hunk 1 above)
Comment (lorentey) [reply validation-test/stdlib/Dictionary.swift]: Hm, maybe StdlibUnittest should do this instead, for every test. However, the relevant file doesn't import Foundation, and I hesitate to add the import there, for fear of even more unintended consequences.

**Comment by lorentey** [comment]
@swift-ci please smoke test linux platform

**Comment by lorentey** [comment]
@swift-ci please test

**Comment by lorentey** [comment]
@swift-ci please test

**Pull request merged**